flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -604,6 +604,9 @@ public void addAllToProperties(Properties props) {
     public void addAll(Configuration other) {
         synchronized (this.confData) {
             synchronized (other.confData) {
+                for (String key : other.keySet()) {
```

Comment:
This class is used by a lot of Flink components has been proven to be relatively mature. I would like to avoid changes here.